robotiq_driver/src/robotiq_gripper_interface.cpp

```diff
@@ -80,40 +81,49 @@ RobotiqGripperInterface::RobotiqGripperInterface(const std::string& com_port, ui
 {
   if (!port_.isOpen())
   {
-    std::cerr << "Failed to open gripper port.\n";
-    return;
+    throw std::runtime_error("Failed to open gripper port.");
   }
 }
 
-bool RobotiqGripperInterface::activateGripper()
+void RobotiqGripperInterface::activateGripper()
 {
-  auto cmd = createWriteCommand(kActionRequestRegister, { 0x0100, 0x0000, 0x0000 }  // set rACT to 1, clear all
+  const auto cmd = createWriteCommand(kActionRequestRegister, { 0x0100, 0x0000, 0x0000 }  // set rACT to 1, clear all
                                                                                     // other registers.
   );
-  if (!sendCommand(cmd)) {
-    return false;
-  }
 
-  readResponse(kWriteResponseSize);
+  try{
+    sendCommand(cmd);
+    readResponse(kWriteResponseSize);
 
-  updateStatus();
-  while (gripper_status_ == GripperStatus::IN_PROGRESS)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
     updateStatus();
+    
+    if(gripper_status_ == GripperStatus::COMPLETED){
+      return;
+    }
+
+    while (gripper_status_ == GripperStatus::IN_PROGRESS)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+      updateStatus();
+    }
```

Comment:
Interestingly enough, this is now the only bit of code where the opening brace `{` is on a new line. No big deal, but maybe for consistency bring it up to the previous line?